File: php/09proc/proc59.php
<?php
function Leng($xA, $yA, $xB, $yB){
        return number_format(sqrt(pow($xA - $xB, 2) + pow($yA - $yB, 2)), 2);
}
function Area($xA, $yA, $xB, $yB, $xC, $yC){
        $a = Leng($xA, $yA, $xB, $yB);
        $b = Leng($xB, $yB, $xC, $yC);
        $c = Leng($xA, $yA, $xC, $yC);
        $P = $a + $b + $c;
        $p = $P / 2;
        $S = sqrt($p * ($p - $a) * ($p - $b) * ($p - $c));
		$S = number_format($S, 2);
        return $S;
}
function Dist($xP, $yP, $xA, $yA, $xB, $yB){
        return number_format(2 * Area($xP, $yP, $xA, $yA, $xB, $yB) / Leng($xA, $yA, $xB, $yB), 2);
}
$xP = mt_rand(1, 100);
$yP = mt_rand(1, 100);
$xA = mt_rand(1, 100);
$yA = mt_rand(1, 100);
$xB = mt_rand(1, 100);
$yB = mt_rand(1, 100);
$xC = mt_rand(1, 100);
$yC = mt_rand(1, 100);
echo "xP = ".$xP;
echo "<br>yP = ".$yP;
echo "<br>xA = ".$xA;
echo "<br>yA = ".$yA;
echo "<br>xB = ".$xB;
echo "<br>yB = ".$yB;
echo "<br>xC = ".$xC;
echo "<br>yC = ".$yC;
$PAB = Dist($xP, $yP, $xA, $yA, $xB, $yB);
$PAC = Dist($xP, $yP, $xA, $yA, $xC, $yC);
$PBC = Dist($xP, $yP, $xB, $yB, $xC, $yC);
echo "<br>Dist(P, AB) = ".$PAB;
echo "<br>Dist(P, AC) = ".$PAC;
echo "<br>Dist(P, BC) = ".$PBC;
?>

File: cpp/09proc/Proc59.cpp
#include <iostream>
#include <cmath>
using namespace std;

struct Point
{
	double x, y;
};

double Leng(double xA, double yA, double xB, double yB);
double Area(double xA, double yA, double xB, double yB, double xC, double yC);
double Perim(double xA, double yA, double xB, double yB, double xC, double yC,
			 double& a, double& b, double& c);
double Dist(double xP, double yP, double xA, double yA, double xB, double yB);

int main()
{
	//Task("Proc59");
	Point P, A, B, C;

	cin >> P.x >> P.y;
	cin >> A.x >> A.y;
	cin >> B.x >> B.y;
	cin >> C.x >> C.y;

	cout << Dist(P.x, P.y, A.x, A.y, B.x, B.y);
	cout << Dist(P.x, P.y, A.x, A.y, C.x, C.y);
	cout << Dist(P.x, P.y, B.x, B.y, C.x, C.y);
	
	return 0;
}

double Leng(double xA, double yA, double xB, double yB)
{
	return sqrt( pow(xB - xA, 2) + pow(yB - yA, 2) );
}

double Area(double xA, double yA, double xB, double yB, double xC, double yC)
{
	double a, b, c;
	double p = Perim(xA, yA, xB, yB, xC, yC, a, b, c) / 2;

	return sqrt( p*(p-a)*(p-b)*(p-c) );
}

double Perim(double xA, double yA, double xB, double yB, double xC, double yC,
			 double& a, double& b, double& c)
{
	a = Leng(xA, yA, xB, yB);
	b = Leng(xA, yA, xC, yC);
	c = Leng(xB, yB, xC, yC);

	return a+b+c;
}

double Dist(double xP, double yP, double xA, double yA, double xB, double yB)
{
	return 2 * Area(xP, yP, xA, yA, xB, yB) / Leng(xA, yA, xB, yB);
}